### ISSUE TITLE
/edh Flag Launches Odyssey Instead of Horizons

### ISSUE THREAD

**Issue by francismanjares** [Rfvgyhn/min-ed-launcher#25]
So odyssey has launched. Since the new expansion has so many stuff going on I tried switching to Horizons but /edh is not working for some reason it kept on bringing edo.

thank you.

**Comment by francismanjares** [comment]
ok... removing the arguments let me select which one to launch. but still

**Comment by Rfvgyhn** [comment]
It sounds like the products might have different `filter` values from the `purchases` request. It's possible FDev changed the purchases response values if you own odyssey. I've added a bit more logging that would verify that. Can you try out [this CI build](https://github.com/Rfvgyhn/min-ed-launcher/actions/runs/862246369) and paste the log contents of the last run (`logs/min-ed-launcher.log).

**Comment by francismanjares** [comment]
Here is the logs from the CLI build.

`2021-05-22 04:24:16.141 +08:00 [INF] Elite Dangerous: Minimal Launcher - v0.4.0-ci862246369
2021-05-22 04:24:16.269 +08:00 [DBG] 
    Args: [|"/Epic"; "-AUTH_LOGIN=unused";
  "-AUTH_PASSWORD=6d5...baa"; "-AUTH_TYPE=exchangecode";
  "-epicapp=9c2...593"; "-epicenv=Prod"; "-EpicPortal";
  "/autorun"; "/autoquit"; "/EDH"; "-epicusername=ArchelausHeron";
  "-epicuserid=07a...4a7"; "-epiclocale=en"|]
    OS: Win64
    
2021-05-22 04:24:16.293 +08:00 [DBG] Reading settings from 'C:\Users\franc\AppData\Local\min-ed-launcher\settings.json'
2021-05-22 04:24:16.582 +08:00 [DBG] Settings: Ok { Platform = Epic { ExchangeCode = "6d5...baa"
                       Type = "exchangecode"
                       AppId = "9c2...593" }
     DisplayMode = Pancake
     AutoRun = true
     AutoQuit = true
     WatchForCrashes = false
     ProductWhitelist = set ["edh"]
     ForceLocal = true
     Proton = None
     CbLauncherDir = "H:\EliteDangerous\"
     PreferredLanguage = None
     ApiUri = https://api.zaonce.net/
     Restart = None
     AutoUpdate = true
     MaxConcurrentDownloads = 4
     ForceUpdate = set []
     Processes = [] }
2021-05-22 04:24:16.592 +08:00 [INF] Elite Runtime
    Platform: Epic
    CobraBay Version: 0.4.6705.0
    Products Dir: H:\EliteDangerous\Products
2021-05-22 04:24:17.821 +08:00 [DBG] Requesting epic token
2021-05-22 04:24:19.098 +08:00 [DBG] Requesting epic token success
2021-05-22 04:24:19.106 +08:00 [DBG] Authenticating via Epic
2021-05-22 04:24:19.419 +08:00 [DBG] Successfully authenticated
2021-05-22 04:24:19.422 +08:00 [INF] Logged in via Epic as: Francis Manjares
2021-05-22 04:24:19.422 +08:00 [DBG] Getting authorized products
2021-05-22 04:24:20.766 +08:00 [DBG] Purchases Response:
{"purchases":[{"colour":"#f07b05","filter":"ed","directory":"elite-dangerous-64","serverargs":"","gameargs":"SeasonOne","sortkey":"06","product_name":"Elite Dangerous","product_sku":"FORC-FDEV-D-1010","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#f07b05","filter":"eda","directory":"elite-dangerous-64","serverargs":"","gameargs":"CQCArena","sortkey":"20","product_name":"Elite Dangerous: Arena","product_sku":"FORC-FDEV-D-1012","template":"http://hosting.zaonce.net/launcher/elite-dangerous-arena/en.html"},{"colour":"#f07b05","filter":"ed","directory":"COMBAT_TUTORIAL_DEMO","serverargs":"","gameargs":"","sortkey":"30","product_name":"Single Player Combat Training","product_sku":"COMBAT_TUTORIAL_DEMO","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#0a8bd6","filter":"edh","directory":"elite-dangerous-64","serverargs":"","gameargs":"","sortkey":"05","product_name":"Elite Dangerous: Horizons","product_sku":"FORC-FDEV-D-1013","template":"http://hosting.zaonce.net/launcher/elite-dangerous-horizons/en.html"},{"colour":"#0a8bd6","filter":"edh","directory":"elite-dangerous-odyssey-64","serverargs":"","gameargs":"","sortkey":"00","product_name":"Elite Dangerous: Odyssey","product_sku":"FORC-FDEV-DO-1000","template":"http://hosting.zaonce.net/launcher/odyssey/en.html"}]}
2021-05-22 04:24:20.782 +08:00 [DBG] Authorized Products: Elite Dangerous: Odyssey,Elite Dangerous: Horizons,Elite Dangerous,Elite Dangerous: Arena,Single Player Combat Training
2021-05-22 04:24:20.782 +08:00 [INF] Checking for updates
2021-05-22 04:24:20.787 +08:00 [DBG] Disabling 'Single Player Combat Training'. Unable to find product at 'H:\EliteDangerous\Products\COMBAT_TUTORIAL_DEMO\VersionInfo.txt'
2021-05-22 04:24:25.500 +08:00 [INF] Available Products:
	Elite Dangerous: Odyssey  FORC-FDEV-DO-1000 Up to Date
	Elite Dangerous: Horizons FORC-FDEV-D-1013  Up to Date
	Elite Dangerous           FORC-FDEV-D-1010  Up to Date
	Elite Dangerous: Arena    FORC-FDEV-D-1012  Up to Date
2021-05-22 04:24:26.000 +08:00 [INF] Launching Elite Dangerous: Odyssey
2021-05-22 04:24:26.014 +08:00 [DBG] Process: H:\EliteDangerous\Products\elite-dangerous-odyssey-64\EliteDangerous64.exe "ServerToken ad0...b7a8c2...4e0570...b6c4507b32267c26d0fc550 a5d...a48331...6b0860...8b859b...d0e11c...7e3b4d...e1708c...0a084f...c098c4...1a96f0...fe18a5...ea04fa...9a7a76...3bd6e0...d67ba9...5f5d58...a850ab...5b77bc...362785...19183f...2d7978...93047c...69fcee...4a20fd...931386...7db897...52d06b...0ea68e...062990...f1f6dc...aae676...b4af43...f32f5f...92c17c...294981...ee9135...02025d...d36986...ffc973...08c561...fdf336...c12922f61d7d3 " /language English\\UK "EpicToken eyJ...vRw" /novr
2021-05-22 04:25:27.902 +08:00 [INF] Shutdown Elite Dangerous: Odyssey
`

**Comment by francismanjares** [comment]
> 2021-05-22 04:24:16.141 +08:00 [INF] Elite Dangerous: Minimal Launcher - v0.4.0-ci862246369
2021-05-22 04:24:16.269 +08:00 [DBG] 
    Args: [|"/Epic"; "-AUTH_LOGIN=unused";
  "-AUTH_PASSWORD=6d5...baa"; "-AUTH_TYPE=exchangecode";
  "-epicapp=9c2...593"; "-epicenv=Prod"; "-EpicPortal";
  "/autorun"; "/autoquit"; "/EDH"; "-epicusername=ArchelausHeron";
  "-epicuserid=07a...4a7"; "-epiclocale=en"|]
    OS: Win64
    
2021-05-22 04:24:16.293 +08:00 [DBG] Reading settings from 'C:\Users\franc\AppData\Local\min-ed-launcher\settings.json'
2021-05-22 04:24:16.582 +08:00 [DBG] Settings: Ok { Platform = Epic { ExchangeCode = "6d5...baa"
                       Type = "exchangecode"
                       AppId = "9c2...593" }
     DisplayMode = Pancake
     AutoRun = true
     AutoQuit = true
     WatchForCrashes = false
     ProductWhitelist = set ["edh"]
     ForceLocal = true
     Proton = None
     CbLauncherDir = "H:\EliteDangerous\"
     PreferredLanguage = None
     ApiUri = https://api.zaonce.net/
     Restart = None
     AutoUpdate = true
     MaxConcurrentDownloads = 4
     ForceUpdate = set []
     Processes = [] }
2021-05-22 04:24:16.592 +08:00 [INF] Elite Runtime
    Platform: Epic
    CobraBay Version: 0.4.6705.0
    Products Dir: H:\EliteDangerous\Products
2021-05-22 04:24:17.821 +08:00 [DBG] Requesting epic token
2021-05-22 04:24:19.098 +08:00 [DBG] Requesting epic token success
2021-05-22 04:24:19.106 +08:00 [DBG] Authenticating via Epic
2021-05-22 04:24:19.419 +08:00 [DBG] Successfully authenticated
2021-05-22 04:24:19.422 +08:00 [INF] Logged in via Epic as: Francis Manjares
2021-05-22 04:24:19.422 +08:00 [DBG] Getting authorized products
2021-05-22 04:24:20.766 +08:00 [DBG] Purchases Response:
{"purchases":[{"colour":"#f07b05","filter":"ed","directory":"elite-dangerous-64","serverargs":"","gameargs":"SeasonOne","sortkey":"06","product_name":"Elite Dangerous","product_sku":"FORC-FDEV-D-1010","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#f07b05","filter":"eda","directory":"elite-dangerous-64","serverargs":"","gameargs":"CQCArena","sortkey":"20","product_name":"Elite Dangerous: Arena","product_sku":"FORC-FDEV-D-1012","template":"http://hosting.zaonce.net/launcher/elite-dangerous-arena/en.html"},{"colour":"#f07b05","filter":"ed","directory":"COMBAT_TUTORIAL_DEMO","serverargs":"","gameargs":"","sortkey":"30","product_name":"Single Player Combat Training","product_sku":"COMBAT_TUTORIAL_DEMO","template":"http://hosting.zaonce.net/launcher/elite/en.html"},{"colour":"#0a8bd6","filter":"edh","directory":"elite-dangerous-64","serverargs":"","gameargs":"","sortkey":"05","product_name":"Elite Dangerous: Horizons","product_sku":"FORC-FDEV-D-1013","template":"http://hosting.zaonce.net/launcher/elite-dangerous-horizons/en.html"},{"colour":"#0a8bd6","filter":"edh","directory":"elite-dangerous-odyssey-64","serverargs":"","gameargs":"","sortkey":"00","product_name":"Elite Dangerous: Odyssey","product_sku":"FORC-FDEV-DO-1000","template":"http://hosting.zaonce.net/launcher/odyssey/en.html"}]}
2021-05-22 04:24:20.782 +08:00 [DBG] Authorized Products: Elite Dangerous: Odyssey,Elite Dangerous: Horizons,Elite Dangerous,Elite Dangerous: Arena,Single Player Combat Training
2021-05-22 04:24:20.782 +08:00 [INF] Checking for updates
2021-05-22 04:24:20.787 +08:00 [DBG] Disabling 'Single Player Combat Training'. Unable to find product at 'H:\EliteDangerous\Products\COMBAT_TUTORIAL_DEMO\VersionInfo.txt'
2021-05-22 04:24:25.500 +08:00 [INF] Available Products:
	Elite Dangerous: Odyssey  FORC-FDEV-DO-1000 Up to Date
	Elite Dangerous: Horizons FORC-FDEV-D-1013  Up to Date
	Elite Dangerous           FORC-FDEV-D-1010  Up to Date
	Elite Dangerous: Arena    FORC-FDEV-D-1012  Up to Date
2021-05-22 04:24:26.000 +08:00 [INF] Launching Elite Dangerous: Odyssey
2021-05-22 04:24:26.014 +08:00 [DBG] Process: H:\EliteDangerous\Products\elite-dangerous-odyssey-64\EliteDangerous64.exe "ServerToken ad0...b7a8c2...4e0570...b6c4507b32267c26d0fc550 a5d...a48331...6b0860...8b859b...d0e11c...7e3b4d...e1708c...0a084f...c098c4...1a96f0...fe18a5...ea04fa...9a7a76...3bd6e0...d67ba9...5f5d58...a850ab...5b77bc...362785...19183f...2d7978...93047c...69fcee...4a20fd...931386...7db897...52d06b...0ea68e...062990...f1f6dc...aae676...b4af43...f32f5f...92c17c...294981...ee9135...02025d...d36986...ffc973...08c561...fdf336...c12922f61d7d3 " /language English\\UK "EpicToken eyJ...vRw" /novr
2021-05-22 04:25:27.902 +08:00 [INF] Shutdown Elite Dangerous: Odyssey

**Comment by francismanjares** [comment]
sorry accidentally clicked closed XD

**Comment by Rfvgyhn** [comment]
```json
{
   "colour":"#0a8bd6",
   "filter":"edh",
   "directory":"elite-dangerous-odyssey-64",
   "serverargs":"",
   "gameargs":"",
   "sortkey":"00",
   "product_name":"Elite Dangerous: Odyssey",
   "product_sku":"FORC-FDEV-DO-1000",
   "template":"http://hosting.zaonce.net/launcher/odyssey/en.html"
}
```

Ugh, it looks like they didn't update the filter key. During the alpha, it was `edo`. I don't know if this is just a copy/paste error on their part or if they intend for horizons and odyssey to have the same filter.

I'm hesitant to fix this by hard-coding filters as that could break compatibility with the default launcher in the future. I think, for now, I'll consider this a bug on their end. If it turns out this is intended behavior, I'll look into breaking compatibility and hard-coding the filters.

You can leave this open for now. Hopefully this is just an oversight on their part and it will get changed in one of their hotfixes. I've opened an [issue](https://issues.frontierstore.net/issue-detail/33154) on their tracker.

**Comment by francismanjares** [comment]
*facepalm* frontier lol. I can wait for te fix , for now I can use this to switch edh and edo quickly :) thank you. o7

**Comment by Rfvgyhn** [comment]
Just wanted to add, I don't own Odyssey so I can't actually check if this one is fixed on their end. If anyone happens to notice `/edh` works again or can check their logs every now and then and report if it has been fixed, that would be great.

**Comment by Rfvgyhn** [comment]
Well, since FDev hasn't fixed this yet, I added the ability to override filters via the settings file. You can add the following property to `settings.json` to fix Odyssey's filter as of [v0.5.0](https://github.com/Rfvgyhn/min-ed-launcher/releases/tag/v0.5.0).

```json
"filterOverrides": [
    { "sku": "FORC-FDEV-DO-1000", "filter": "edo" }
  ]
```
